actions/views/channel.js

```diff
@@ -149,6 +149,15 @@ export function leaveChannel(channelId) {
     };
 }
 
+export function leaveDirectChannel() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const currentUserId = getCurrentUserId(state);
+        const currentTeam = getCurrentTeam(state);
+        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeam.id, state);
```

Comment:
I'm confused how this works, how does this remove the user from a DM channel? It looks like they would just get re-added on refresh.

Reply:
maybe i missread this, but if I understood it correctly is like this: if I leave a DM i get the last channel I accessed, whether is the one I'm in or not. is this right? I think we should only remove it if I was really in that channel, we would also need to take a look at the penultimate one (in case then i go and remove another one and I'm back to the dm which i shouldn't). Just to try and clarify I thought of two weird case scenarios:
- **case 1**: go to dm A, go to dm B, close dm A, reload. Haven't tested but it looks like I'd be back at A
- **case 2**: go to dm A, go to dm B, close dm C, reload. I think we are redirected to dm A.

Reply:
@Willyfrog the first scenario worked fine but in the second one I can't close dm C if I have clicked on dm A and B before so I have to check the penultimate like you said. Thanks